src/libos/src/vm/vm_manager.rs

```diff
@@ -310,22 +338,19 @@ impl VMManager {
 
         intersect_chunks.iter().for_each(|chunk| {
             if let ChunkType::SingleVMA(vma) = chunk.internal() {
-                if let Some(intersection_range) = chunk.range().intersect(&reset_range) {
-                    let mut internal_manager = self.internal();
-                    internal_manager.mprotect_single_vma_chunk(
-                        &chunk,
-                        intersection_range,
-                        VMPerms::DEFAULT,
-                    );
-
-                    unsafe {
-                        let buf = intersection_range.as_slice_mut();
-                        buf.iter_mut().for_each(|b| *b = 0)
-                    }
+                let mut vma = vma.lock().unwrap();
+                if let Some(intersection_vma) = vma.intersect(&reset_range) {
```

Comment:
Due to the introduction of `PageTracker` in VMA, the` subtract`/`intersect` operation of VMA will become more expensive. I suggest review functions involving these operations for optimization.


vm_manger.rs:

- mprotect_single_vma_chunk
- munmap_chunk
- reset_memory

vm_chunk_manager.rs:
- mprotect
- munmap_range

Reply:
Thank you for your review. I take a look at all the calling points, it looks like we can't change `vma intersect/subtract` to `VMRange intersect/subtract`. However, I try to optimize the base function of `vma intersect/subtract`, which is `inherits_file_from`. Previously, `inherits_file_from` needs to go to the global page chunk to update pages for the new vma. In the new implementation, it directly splits the `pages` to fit its range.